Add a way to render tiled layers that are nested within at least one group layer

BloodLantern opened the pull request:
Following my issue, here is the code change to allow using nested layers for rendering, fixing #892.

prime31 commented:
Looks good, the only issue is that it is a big breaking change for all existing projects. I posted the PR in the discord to start a discussion. One quick and easy alternative would be to just have a new component with the `public ITmxLayer[] LayersToRender` to avoid breaking any existing code.

BloodLantern commented:
I just realized that you can put slashes in a tiled layer name, so actually I don't know if it would work that well

BloodLantern commented:
I implemented the path-like syntax to get layers with slashes or any given separator.
The way I implemented the function makes it so that it shouldn't break any existing project (except for the API change for LayersToRender) because it first checks, just like it was doing before, for a layer with exact name matching.

BloodLantern commented:
@prime31 should I change something or is this ready to be merged ?

BloodLantern commented:
I added the `SetLayerIndicesToRender(params int[])` function I suggested above to make it easy for users of `LayerIndicesToRender` to switch to the new API.
